### PR TITLE
ARIA selection mixin

### PR DESCRIPTION
**Pull request by JanMiksovsky**
[Full text of the proposal](https://github.com/JanMiksovsky/rfcs/blob/selection-aria/proposals/0000-selection-aria.md)

This proposes a mixin called `SelectionAriaMixin` for list-like components that want to expose their selection to screen readers and other assistive technologies via [ARIA](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA) accessibility attributes.

### CODE REVIEW

**Comment by jouni** [comment]
Nit: 

> “a default tabindex value.)” 

Unnecessary bracket.

Otherwise LGTM

**Comment by JanMiksovsky** [comment]
Thanks for the attention to detail. Fixed.

File: proposals/0000-selection-aria.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+Suppose the developer initially populates the DOM as follows:
+
+    <accessible-list aria-label="Fruits" tabindex="0">
+      <div>Apple</div>
+      <div>Banana</div>
+      <div>Cherry</div>
+    </accessible-list>
+
+After the element is added to the page, and the first item is selected. The
```
Comment (robdodson) [inline proposals/0000-selection-aria.md]: Based on the example the `role` correctly gets applied before the first item is selected but the wording of this makes it seem like it wouldn't happen until after the first item is selected

File: proposals/0000-selection-aria.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+      <div role="option" id="_option0" aria-selected="true">Apple</div>
+      <div role="option" id="_option1" aria-selected="false">Banana</div>
+      <div role="option" id="_option2" aria-selected="false">Cherry</div>
+    </accessible-list>
+
+The `SelectionAriaMixin` has selected appropriate default values for the
+attributes `role`, `id`, `aria-activedescendant`, and `aria-selected`.
+
+In practice, some additional attributes must be set for ARIA to be useful. The
+`aria-label` attribute is context-dependent and must be set by the author. In
```
Comment (robdodson) [inline proposals/0000-selection-aria.md]: Might also mention `aria-labelledby`

**Comment by robdodson** [comment]
I really like this idea and would love to see it continued!

One thing that I don't think was mentioned in the proposal was how the mixin interacts with keyboard behavior. I'm assuming the mixin does not add any of its own event listeners (I figure it takes advantage of the property setters, right?) but you may want to explicitly call that out.

**Comment by JanMiksovsky** [comment]
@robdodson Many thanks for your expert eye on this RFC. I've incorporated your feedback on all points above.

**Pull request merged**